optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/websocket/RoutePublisherImpl.java

```diff
@@ -47,14 +47,18 @@ public void publish(RoutingPlan routingPlan) {
     }
 
     static PortableRoutingPlan portable(RoutingPlan routingPlan) {
+        List<PortableVehicle> vehicles = routingPlan.vehicles().stream()
+                .map(PortableVehicle::fromVehicle)
+                .collect(Collectors.toList());
         PortableLocation depot = routingPlan.depot().map(PortableLocation::fromLocation).orElse(null);
         List<PortableRoute> routes = routingPlan.routes().stream()
                 .map(routeWithTrack -> new PortableRoute(
+                        PortableVehicle.fromVehicle(routeWithTrack.vehicle()),
                         depot,
                         portableVisits(routeWithTrack.visits()),
                         portableTrack(routeWithTrack.track())))
                 .collect(Collectors.toList());
-        return new PortableRoutingPlan(routingPlan.distance(), depot, routes);
+        return new PortableRoutingPlan(routingPlan.distance(), vehicles, depot, routes);
```

Comment:
As we discussed on gchat, there might be a factory class for each non-trivial domain class. For example here, a factory method taking `RoutingPlan` and creating a new instance of `PortableRoutingPlan` could be handy.

Reply:
Good suggestion. I'll address this in a subsequent PR. Leaving this conversation open until then.

Reply:
Filed #148.